### PR TITLE
Workaround voice disconnect at switch channel

### PR DESCRIPTION
**Pull request by james58899**
Discord using same close code on kick and change voice server (switch channel will change voice server).
We need wait `VOICE_SERVER_UPDATE` to know it is kick or change voice server.

This will fix 4014 and 4006 cause connection disconnect but will cause `disconnect` event delay 5s.

Workaround https://github.com/discord/discord-api-docs/issues/2450

### CODE REVIEW

File: lib/voice/VoiceConnection.js [hunk 1]
```diff
@@ -115,6 +115,7 @@ class VoiceConnection extends EventEmitter {
         this.sequence = 0;
         this.timestamp = 0;
         this.ssrcUserMap = {};
+        this.disconnectTimeout;
```
Comment (bsian03) [inline lib/voice/VoiceConnection.js]: Prefix this maybe? Only nitpick I have

**Review by bsian03 — COMMENTED** [review]
Discord gotta break stuff

File: lib/voice/VoiceConnection.js (hunk 1 above)
Comment (abalabahaha) [reply lib/voice/VoiceConnection.js]:
```suggestion
        this.disconnectTimeout = null;
```

File: lib/voice/VoiceConnection.js [hunk 2]
```diff
@@ -388,14 +389,22 @@ class VoiceConnection extends EventEmitter {
...
+                    this.channelID = null;
+                    this.updateVoiceState();
+                    /**
+                    * Fired when the voice connection disconnects
+                    * @event VoiceConnection#disconnect
+                    * @prop {Error?} error The error, if any
+                    */
+                    this.emit("disconnect", error);
+                }
+            }, 5000);
```
Comment (abalabahaha) [inline lib/voice/VoiceConnection.js]:
```suggestion
            }, this.shard.client ? this.shard.client.options.connectionTimeout : 30000);
```

**Review by abalabahaha — DISMISSED** [review]

**Comment by abalabahaha** [comment]
On second thought, I think it would be better to just treat this like a new connection (but keeping most state). I have some test commits on dev branch, mind stopping by on Discord?

**Comment by james58899** [comment]
Tested dev branch, everything is good.

**Pull request closed without merging**